src/plugins/query_enhancements/server/routes/resources/routes.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { IRouter } from 'opensearch-dashboards/server';
+import { BASE_API } from '../../../common';
+import { resourceManagerService } from '../../connections/resource_manager_service';
+import { coerceStatusCode } from '..';
+
+export function registerResourceRoutes(router: IRouter) {
+  router.post(
+    {
+      path: `${BASE_API}/resources`,
+      validate: {
+        body: schema.object({
+          connection: schema.object({
+            id: schema.string(),
+            type: schema.string(),
+          }),
+          resource: schema.object({
+            name: schema.maybe(schema.string()),
+            type: schema.string(),
+          }),
+          content: schema.maybe(schema.object({}, { unknowns: 'allow' })),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const { type } = request.body.connection;
+      const manager = resourceManagerService.getManager(type);
+      if (!manager) {
+        return response.notFound();
+      }
+      try {
+        const resourcesResponse = await manager.getResources(context, request);
```

Comment:
i just noticed that this is still `getResources`. with this how would you create a resource? while you may pass in arbitrary operations as resource type, it won't make sense if the manager call is always `getResources`

it might be better to make the path `${BASE_API}/resources/search` (i would prefer to have a separate POST route for search, rather than putting get, list, create in the same route), or change this to `handlePostRequest` rather than `getResources`, in order to distinguish between get and create